public/admin.php

```diff
@@ -92,6 +98,43 @@
     return back()->withErrors(__('legacy.error.error'));
 }
 
+if ($action === 'migrate-achievement') {
+    $achievementIds = explode(',', requestInputSanitized('a'));
+    $gameId = requestInputSanitized('g');
+    if (!Game::where('ID', $gameId)->exists()) {
+        return back()->withErrors('Unknown game');
+    }
+
+    // determine which game(s) the achievements are coming from
+    $oldGames = Achievement::whereIn('ID', $achievementIds)->select(['GameID'])->distinct()->pluck('GameID')->toArray();
+
+    // associate the achievements to the new game
+    Achievement::whereIn('ID', $achievementIds)->update(['GameID' => $gameId]);
+
+    // add an audit comment to the new game
+    addArticleComment('Server', ArticleType::GameModification, $gameId,
+        "$user migrated " . Str::plural('achievement', count($achievementIds)) . ' ' .
+        implode(',', $achievementIds) . ' from ' .
+        Str::plural('game', count($oldGames)) . ' ' . implode(',', $oldGames) . '.');
+
+    // ensure player_game entries exist for the new game for all affected users
+    foreach (PlayerAchievement::whereIn('achievement_id', $achievementIds)->select(['user_id'])->distinct()->pluck('user_id')->toArray() as $userId) {
```

Comment:
`->toArray()` shouldn't be necessary as Collections implement the necessary iterator and ArrayAccess interfaces and can be used in `foreach` loops directly.